subscriber/simple_supply_subscriber/database.py

```diff
@@ -162,3 +168,105 @@ def disconnect(self):
         LOGGER.info('Disconnecting from database')
         if self._conn is not None:
             self._conn.close()
+
+    def commit(self):
+        self._conn.commit()
+
+    def drop_fork(self, block_num):
+        """Deletes all resources from a particular block_num
+        """
+        delete_agents = """
+        DELETE FROM agents WHERE start_block_num >= {}
+        """.format(block_num)
+        update_agents = """
+        UPDATE agents SET end_block_num = null
+        WHERE end_block_num >= {}
+        """.format(block_num)
+
+        delete_record_locations = """
+        DELETE FROM record_owners WHERE record_id =
+        (SELECT record_id FROM records WHERE start_block_num >= {})
+        """.format(block_num)
+        delete_record_owners = """
+        DELETE FROM record_owners WHERE record_id =
+        (SELECT record_id FROM records WHERE start_block_num >= {})
+        """.format(block_num)
+        delete_records = """
+        DELETE FROM records WHERE start_block_num >= {}
+        """.format(block_num)
+        update_records = """
+        UPDATE records SET end_block_num = null
+        WHERE end_block_num >= {}
+        """.format(block_num)
+
+        delete_blocks = """
+        DELETE FROM blocks WHERE block_num >= {}
+        """.format(block_num)
+
+        with self._conn.cursor() as cursor:
+            cursor.execute(delete_agents)
+            cursor.execute(update_agents)
+            cursor.execute(delete_record_locations)
+            cursor.execute(delete_record_owners)
+            cursor.execute(delete_records)
+            cursor.execute(update_records)
+            cursor.execute(delete_blocks)
+
+        self._conn.commit()
```

Comment:
This should still be part of a single transaction - one event should equal one PostGres transaction.